functions/Export-DbaScript.ps1

```diff
@@ -213,75 +233,80 @@ function Export-DbaScript {
+                                $scriptpart  | Out-String
                             }
-                            $scriptpart  | Out-String
                         }
                     } else {
                         if ($ScriptingOptionsObject) {
-                            if ($ScriptingOptionsObject.ScriptBatchTerminator) {
-                                $ScriptingOptionsObject.AppendToFile = $true
+                            if ($scriptBatchTerminator) {
+                                # Option to script batch terminator via ScriptingOptionsObject needs to write to file only
+                                $ScriptingOptionsObject.AppendToFile = (($null -ne $prefix) -or $appendToScript )
                                 $ScriptingOptionsObject.ToFileOnly = $true
                                 if (-not  $ScriptingOptionsObject.FileName) {
-                                    $ScriptingOptionsObject.FileName = $FilePath
+                                    $ScriptingOptionsObject.FileName = $scriptPath
                                 }
-                                $object.Script($ScriptingOptionsObject)
+                                $null = $object.Script($ScriptingOptionsObject)
+                                # Reset the changed values of the $ScriptingOptionsObject in case it is reused later
+                                $ScriptingOptionsObject.AppendToFile = $soAppendToFile
+                                $ScriptingOptionsObject.ToFileOnly = $soToFileOnly
+                                $ScriptingOptionsObject.FileName = $soFileName
                             } else {
+                                $ScriptingOptionsObject.FileName = $null
                                 foreach ($scriptpart in $scripter.EnumScript($object)) {
-                                    if ($BatchSeparator -ne "") {
+                                    if ($scriptBatchTerminator) {
                                         $scriptpart = "$scriptpart`r`n$BatchSeparator`r`n"
                                     }
-                                    $scriptpart | Out-File -FilePath $FilePath -Encoding $encoding -Append
+                                    $scriptpart | Out-File -FilePath $scriptPath -Encoding $encoding -Append
                                 }
+                                $ScriptingOptionsObject.FileName = $soFileName
                             }
-
                         } else {
-                            if (Get-Member -Name ScriptCreate -InputObject $object) {
-                                $scriptpart = $object.ScriptCreate().GetScript()
-                            } else {
-                                $scriptpart = $object.Script()
-                            }
-                            if ($BatchSeparator -ne "") {
-                                $scriptpart = "$scriptpart`r`n$BatchSeparator`r`n"
+                            foreach ($scriptpart in $scripter.EnumScript($object)) {
```

Comment:
This line is within the section that executes when no $ScriptingOptionsObject is passed, but was changed to use $scripter.EnumScript.  Are there other benefits to using this? Working on issue #5930 and the Extended Events object doesn't support EnumScript. Just want to make sure if I submit a PR to change this back that I'm not overlooking something?

Reply:
Hi Jess

I actually have an Export-DbaXeSession command that I am going to submit today that addresses the issue. Scripting extended events uses totally different construct and can't use Export-DbaScript